src/CoherenceBuild/PackagePublisher.cs

```diff
@@ -13,64 +14,26 @@ namespace CoherenceBuild
 {
     public static class PackagePublisher
     {
-        public static void PublishToShare(
-            ProcessResult processResult,
-            string outputPath,
-            string symbolsOutputPath)
+        public static void PublishToFeed(List<PackageInfo> processedPackages, string feed, string apiKey)
         {
-            Directory.CreateDirectory(outputPath);
-            Directory.CreateDirectory(symbolsOutputPath);
-            var packagesToCopy = processResult.AllPackages.Values;
-
-            Parallel.ForEach(packagesToCopy, new ParallelOptions { MaxDegreeOfParallelism = 4 }, packageInfo =>
-            {
-                var packagePath = Path.Combine(outputPath, Path.GetFileName(packageInfo.PackagePath));
-
-                Program.Retry(() =>
-                {
-                    File.Copy(packageInfo.PackagePath, packagePath, overwrite: true);
-                    // Update package path to point to local share.
-                    packageInfo.PackagePath = packagePath;
-                });
-
-                if (File.Exists(packageInfo.SymbolsPath))
-                {
-                    Program.Retry(() =>
-                    {
-                        File.Copy(
-                            packageInfo.SymbolsPath,
-                            Path.Combine(symbolsOutputPath, Path.GetFileName(packageInfo.SymbolsPath)),
-                            overwrite: true);
-                    });
-                }
-
-                Log.WriteInformation("Copied to {0}", packagePath);
-            });
-        }
-
-        public static void PublishToFeed(ProcessResult processResult, string feed, string apiKey)
-        {
-            PublishToFeedAsync(processResult, feed, apiKey).Wait();
+            PublishToFeedAsync(processedPackages, feed, apiKey).Wait();
         }
 
-        private static async Task PublishToFeedAsync(ProcessResult processResult, string feed, string apiKey)
+        private static async Task PublishToFeedAsync(List<PackageInfo> processedPackages, string feed, string apiKey)
         {
-            var packagesToPushInOrder = Enumerable.Concat(
-                 processResult.CoreCLRPackages.Values,
-                 processResult.ProductPackages.OrderBy(p => p.Value.Degree).Select(p => p.Value));
-
+            processedPackages.Sort((a, b) => a.Degree - b.Degree);
             using (var semaphore = new SemaphoreSlim(4))
```

Comment:
This is a lot of ceremony for just uploading packages. If you use a concurrent collection as I mentioned before you can make this way simpler. I'm guessing this code is trying to limit the concurrency through the use of the semaphore slim. I would suggest simplifying this by using https://msdn.microsoft.com/en-us/library/dd783942(v=vs.110).aspx with https://msdn.microsoft.com/en-us/library/system.threading.tasks.paralleloptions.maxdegreeofparallelism(v=vs.110).aspx
or us PLINQ directly with https://msdn.microsoft.com/en-us/library/dd383719(v=vs.110).aspx